src/components/user/newPostDialog.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {useState} from 'react'
+import axios from 'axios'
+import firebase from 'firebase'
+import {
+   makeStyles,
+   Input,
+   Typography,
+   TextField,
+   Dialog,
+   Button
+} from '@material-ui/core'
+
+const useStyles=makeStyles(theme=>({
+   container:{
+      'max-width': '90%'
+   },
+   dialogContainer:{
+      padding: '10px 30px 20px',
+      display: 'flex',
+      flexDirection: 'column',
+   },
+   closeBtn:{
+      margin: '10px 10px 0'
+   },
+   editInput:{
+      margin: '10px 0'
+   },
+   editBtnContainer:{
+      display: 'flex',
+      justifyContent: 'space-between'
+   },
+   fileInput:{
+      margin: '30px 0',
+      padding: '10px'
+   },
+   btnContainer:{
+      display: 'flex',
+      justifyContent: 'space-around'
+   },
+   imagePreview:{
+      maxHeight: '50vh',
+      objectFit: 'cover'
+   }
+}))
+
+export const NewPostDialog=({open, setOpen, setPosts, user}: {
```

Comment:
please add interface to NewPostDialog in separate file and use it here

Reply:
I've created new file with all my interfaces